File: src/main/java/ukalus/persistence/memory/MemoryStore.java
package ukalus.persistence.memory;

import ukalus.persistence.Reference;
import ukalus.persistence.Store;

import java.io.IOException;
import java.io.Serializable;

public class MemoryStore<T extends Serializable> implements Store<T> {

  private Reference<T> reference = new Reference<>();

  public void store(T object) throws IOException {
    reference.set(object);
  }

  public T load() throws IOException, ClassNotFoundException {
    return reference.get();
  }

  public void close() throws IOException {
  }

}

File: src/main/java/ukalus/persistence/memory/MemoryStore.kt
package ukalus.persistence.memory

import ukalus.persistence.Reference
import ukalus.persistence.Store

import java.io.IOException
import java.io.Serializable

class MemoryStore<T : Serializable> : Store<T> {

    private val reference = Reference<T>()

    @Throws(IOException::class)
    override fun store(`object`: T?) {
        reference.set(`object`)
    }

    @Throws(IOException::class, ClassNotFoundException::class)
    override fun load(): T? {
        return reference.get()
    }

    @Throws(IOException::class)
    override fun close() {
    }

}
